xbmc/filesystem/PluginDirectory.cpp

```diff
@@ -70,6 +70,42 @@ bool CPluginDirectory::StartScript(const std::string& strPath, bool resume)
   return RunScript(this, addon, strPath, resume);
 }
 
+bool CPluginDirectory::GetResolvedPluginResult(CFileItem& resultItem)
+{
+  std::string lastResolvedPath;
+  if (resultItem.HasProperty("EnsurePathCallback") &&
+      resultItem.GetProperty("EnsurePathCallback").asBoolean())
+  {
+    // ensures that a plugin have the callback to resolve the paths in any case
+    // also when the same items in the playlist are played more times
+    lastResolvedPath = resultItem.GetPluginPath();
+  }
+  else
+    lastResolvedPath = resultItem.GetDynPath();
+
+  if (!lastResolvedPath.empty())
+  {
+    // we try to resolve recursively up to 5 nested plugin paths to avoid deadlocks
+    //  (plugin:// paths can resolve to plugin:// paths)
+    for (int i = 0; URIUtils::IsPlugin(lastResolvedPath) && i < 5; ++i)
```

Comment:
Since the 5 magic number is now restricted to this class, can you move it to a const value in this file? Anonymous namespace

Reply:
done, take a look on this if is correct